engine/src/main/java/org/hibernate/search/query/dsl/sort/SortFieldContext.java

```diff
@@ -0,0 +1,51 @@
+package org.hibernate.search.query.dsl.sort;
+
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+import java.util.Comparator;
+
+import org.hibernate.search.spatial.Coordinates;
+
+/**
+ * @author Emmanuel Bernard emmanuel@hibernate.org
+ */
+public interface SortFieldContext extends SortAdditionalSortFieldContext, SortOrder<SortFieldContext>, SortTermination {
+
+	// Geospatial sorting
+
+	/**
+	 * Order elements by their distance from {@code coordinates}.
+	 */
+	SortDistanceContext fromCoordinates(Coordinates coordinates);
+
+	/**
+	 * Latitude in degree
+	 */
+	SortLatLongContext fromLatitude(double latitude);
+
+	// native comparator
+
+	/**
+	 * Compare field with specific comparator implementation.
+	 */
+	SortComparatorContext withComparator(Comparator<?> comparator);
```

Comment:
It might be harder than I hoped so we will likely drop it. I pjust wanted to show all the options before removing them.